src/components/activity-page/managed-interactive/iframe-runtime.tsx

```diff
@@ -174,6 +181,10 @@ export const IframeRuntime: React.FC<IProps> =
   }, [url, authoredState, report, initialInteractiveState, setNewHint, getFirebaseJWT, setSupportedFeatures,
       setSendCustomMessage, showModal, closeModal]);
 
+  useImperativeHandle(ref, () => ({
+    requestInteractiveState: () => phoneRef.current?.post("getInteractiveState")
+  }));
```

Comment:
My reading of the docs suggests that the `forwardRef()`/`useImperativeHandle()` combination is specifically designed to handle the case of refs to DOM elements/React components which can't be passed as regular props due to the special-case handling of `ref` props. At the very least a comment seems warranted explaining how it's being used in this case. I would have thought that you could achieve the same effect by passing in a standard prop like `onSetRequestInteractiveState` but I suspect there's a reason you went with the `forwardRef()`/`useImperativeHandle()` approach.

Reply:
My understanding is that `useImperativeHandle` is a way to provide imperative API that could be done before just by defining a new class method. So, API that is provided by a child component and called by the parent. The opposite of typical callbacks provided via props. An example would be a parent calling `input.focus()` vs `onFocus` called by the chilled itself. 

Here the situation is the same. It's a way for the parent to tell the child (IframeRuntime) to request its state using iframe-phone. It happens when ManagedInteractive component is about to close the dialog with IFrameRuntime.